core/processor/tx.go

```diff
@@ -21,54 +22,53 @@ import (
 	"code.vegaprotocol.io/vega/core/txn"
 	"code.vegaprotocol.io/vega/libs/proto"
 	commandspb "code.vegaprotocol.io/vega/protos/vega/commands/v1"
-	wcrypto "code.vegaprotocol.io/vega/wallet/crypto"
-
 	"github.com/tendermint/tendermint/crypto/tmhash"
 )
 
-var ErrUnsupportedFromValueInTransaction = errors.New("unsupported value from `from` field in transaction")
-
 type Tx struct {
 	originalTx []byte
 	tx         *commandspb.Transaction
 	inputData  *commandspb.InputData
 	err        error
 	pow        *commandspb.ProofOfWork
-	version    uint32
+	version    commandspb.TxVersion
 }
 
 func DecodeTxNoValidation(payload []byte) (*Tx, error) {
 	tx := &commandspb.Transaction{}
 	if err := proto.Unmarshal(payload, tx); err != nil {
 		return nil, fmt.Errorf("unable to unmarshal transaction: %w", err)
 	}
-	input := commandspb.InputData{}
-	if err := proto.Unmarshal(tx.InputData, &input); err != nil {
-		return nil, fmt.Errorf("unable to unmarshal input data: %w", err)
+
+	// We ignore the chain ID, as the null chain is meant to be use for simulation.
+	var rawInputData []byte
+	idx := bytes.IndexByte(tx.InputData, commands.ChainIDDelimiter)
+	if idx == -1 {
+		rawInputData = tx.InputData
+	} else {
+		rawInputData = tx.InputData[idx+1:]
 	}
 
+	inputData := &commandspb.InputData{}
+	err := proto.Unmarshal(rawInputData, inputData)
```

Comment:
behaviour have changed here, the unmarshalling of input data would have returned an error if failing.

Reply:
I think we misunderstood eachother last time

Reply:
I used the `err` to value the the `err` property of the Tx struct, because it was not valued anywhere, so it though that was a mistake to be fixed. Since it is not. I will remove that part.

Reply:
yea I guess, here just the proto unmarshal could have returned an error before.